src/main/java/org/candlepin/pki/certs/SCACertificateGenerator.java

```diff
@@ -58,216 +65,333 @@
 
-        KeyPair keyPair = this.keyPairGenerator.getKeyPair(consumer);
-        org.candlepin.model.dto.Product container = createSCAProdContainer(owner, consumer);
+        if (!v3CapabilityCheck.isCertV3Capable(consumer)) {
+            log.debug("Consumer is not v3 cert capable");
+            return null;
+        }
 
-        List<Environment> environments = this.environmentCurator.getConsumerEnvironments(consumer);
-        ContentPathBuilder contentPathBuilder = ContentPathBuilder.from(owner, environments);
-        PromotedContent promotedContent = new PromotedContent(contentPathBuilder).withAll(environments);
+        // Impl note:
+        // These need to be ordered according to priority! At the time of writing, getConsumerEnvironments
+        // does this, but if that ever changes, we absolutely need that sorting here.
+        List<Environment> environments = environmentCurator.getConsumerEnvironments(consumer);
 
-        Function<ProductContent, String> cidFetcher = pcinfo -> pcinfo.getContent().getId();
+        SCACertificate cert = getCertificate(owner, consumer, environments);
+        ContentAccessPayload payload = getContentAccessPayload(owner, consumer, environments);
 
-        Map<String, ProductContent> ownerContent = this.contentCurator
-            .getActiveContentByOwner(owner.getId())
-            .stream()
-            .collect(Collectors.toMap(cidFetcher, Function.identity(),
-                (v1, v2) -> new ProductContent(v2.getContent(), v1.isEnabled() || v2.isEnabled())));
+        Date certUpdated = cert.getUpdated();
+        Date payloadUpdated = payload.getTimestamp();
 
+        SCACertificate combined = new SCACertificate();
+        combined.setCert(cert.getCert() + payload.getPayload());
+        combined.setCreated(cert.getCreated());
+        combined.setUpdated(payloadUpdated.after(certUpdated) ? payloadUpdated : certUpdated);
+        combined.setId(cert.getId());
+        combined.setKey(cert.getKey());
+        combined.setSerial(cert.getSerial());
 
-        byte[] payloadBytes = createContentAccessDataPayload(consumer, ownerContent, promotedContent);
+        return combined;
+    }
 
-        X509Certificate x509Cert = createX509Cert(consumer.getUuid(), owner, serial,
-            keyPair, container, start, end);
+    private SCACertificate getCertificate(Owner owner, Consumer consumer, List<Environment> environments) {
+        SCACertificate scaCertificate = consumer.getContentAccessCert();
+        if (scaCertificate == null) {
+            log.info("Generating new SCA certificate key for consumer: \"{}\"", consumer.getUuid());
 
-        SCACertificate existing = new SCACertificate();
-        existing.setSerial(serial);
-        existing.setConsumer(consumer);
-        existing.setKeyAsBytes(this.pemEncoder.encodeAsBytes(keyPair.getPrivate()));
-        existing.setCert(this.pemEncoder.encodeAsString(x509Cert));
-        existing.setContent(this.createPayloadAndSignature(payloadBytes));
```

Comment:
Since the payload now lives in a separate entity/table, we should be removing the SCACertificate.content field, and the respective db column

Reply:
Removed the SCACertificate.content field and added a change set to drop the content column from the cp_cont_access_cert table.